headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverJMX.java

```diff
@@ -159,7 +159,7 @@ public SpringProcessMemoryMetricsLiveData refreshMemoryMetrics(SpringProcessLive
 				log.info("retrieve live data from: " + jmxURL);
 				SpringProcessMemoryMetricsLiveData liveData = springJMXConnector.retrieveLiveMemoryMetricsData(getProcessType(), jmxConnection, processID, processName, currentData, metricName, tags);
 				
-				if (liveData != null && liveData.getMemoryMetrics() != null && liveData.getMemoryMetrics().length > 0) {
+				if (liveData != null && liveData.getHeapMemoryMetrics() != null && liveData.getHeapMemoryMetrics().length > 0 && liveData.getNonHeapMemoryMetrics().length > 0) {
```

Comment:
Minor comment. If we are checking liveData.getHeapMemoryMetrics() != null we should do the same for liveData.getNonHeapMemoryMetrics(). Or just don't check for 'null' in both cases